src/bosh_azure_cpi/lib/cloud/azure/disk/disk_manager2.rb

```diff
@@ -167,6 +176,25 @@ def os_disk(vm_name, stemcell_info, size, caching, use_root_disk_as_ephemeral)
       }
     end
 
+    def ephemeral_os_disk(vm_name, stemcell_info, root_disk_size, ephemeral_disk_size, use_root_disk_as_ephemeral, placement)
+      disk_size = get_os_disk_size(root_disk_size, stemcell_info, use_root_disk_as_ephemeral)
+
+      if use_root_disk_as_ephemeral && !ephemeral_disk_size.nil? && root_disk_size.nil?
+        disk_size = get_os_disk_size(stemcell_info.image_size, stemcell_info, use_root_disk_as_ephemeral)
+      end
```

Comment:
If my understanding is correct we want to avoid the minimum_required_disk_size here in case root_disk_size isn't set. Can we make more readable? My suggestion would be  
```
disk_size = if use_root_disk_as_ephemeral && !ephemeral_disk_size.nil? && root_disk_size.nil?
#comment why stemcell_info.image_size
get_os_disk_size(stemcell_info.image_size, stemcell_info, use_root_disk_as_ephemeral)
else
get_os_disk_size(stemcell_info.image_size, stemcell_info, use_root_disk_as_ephemeral)
end

Reply:
Then we use the same logic as before.
The 'or_disk' gets the default size like ever before: 

```
Line 180
disk_size = get_os_disk_size(root_disk_size, stemcell_info, use_root_disk_as_ephemeral
```

But there was no test case for that.
I'll add one